app/views/shelters/show.html.erb

```diff
@@ -1,8 +1,33 @@
-<h1><%= @shelter.name %></h1>
-<p>Rank: <%= @shelter.rank %></p>
-<p>City: <%= @shelter.city %></p>
+<% if @admin %>
 
-<p class="pet-count">Pets at <%= @shelter.name %>: <%= @shelter.pet_count %></p>
-<p><%= link_to "All pets at #{@shelter.name}", "/shelters/#{@shelter.id}/pets" %></p>
+  <h3><%= @shelter %></h3>
+ 
 
-<p><%= link_to "Delete #{@shelter.name}", "/shelters/#{@shelter.id}", method: :delete %></p>
+  <h4> <%= "Action Required" %> </h4>
+  <% @pending_pets.each do |pet| %>
```

Comment:
Here is where I would just have the `@current_shelter` instance variable in your controller and then in this view you could call `@current_shelter.pending_pets` because you should be able to reach through the shelter to get to its pets by the associations that you have.